component/src/main/java/org/wso2/carbon/identity/authenticator/LinkedIn/LinkedInAuthenticator.java

```diff
@@ -0,0 +1,350 @@
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Authenticator of LinkedIn
+ */
+public class LinkedInAuthenticator extends OpenIDConnectAuthenticator implements FederatedApplicationAuthenticator {
+
+    private static Log log = LogFactory.getLog(LinkedInAuthenticator.class);
+
+    /**
+     * check weather user can process or not
+     *
+     * @param request the request
+     * @return true or false
+     */
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+        if (log.isTraceEnabled()) {
+            log.trace("Inside LinkedinOAuth2Authenticator.canHandle()");
+        }
+        // Check commonauth got an OIDC response
+        return request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE) != null
+                && request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_STATE) != null
+                && LinkedInAuthenticatorConstants.LINKEDIN_LOGIN_TYPE.equals(getLoginType(request));
+    }
+
+    private String getLoginType(HttpServletRequest request) {
+        String state = request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_STATE);
+        if (state != null) {
+            return state.split(",")[1];
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Get LinkedIn authorization endpoint.
+     */
+    @Override
+    protected String getAuthorizationServerEndpoint(Map<String, String> authenticatorProperties) {
+        return LinkedInAuthenticatorConstants.LINKEDIN_OAUTH_ENDPOINT;
+    }
+
+    /**
+     * Get LinkedIn token endpoint.
+     */
+    @Override
+    protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
+        return LinkedInAuthenticatorConstants.LINKEDIN_TOKEN_ENDPOINT;
+    }
+
+    /**
+     * Get LinkedIn user info endpoint.
+     */
+    @Override
+    protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        return LinkedInAuthenticatorConstants.LINKEDIN_USERINFO_ENDPOINT;
+    }
+
+    /**
+     * Check ID token in LinkedIn OAuth.
+     */
+    @Override
+    protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
+        return false;
+    }
+
+    /**
+     * Get the friendly name of the Authenticator
+     */
+    @Override
+    public String getFriendlyName() {
+        return LinkedInAuthenticatorConstants.LINKEDIN_CONNECTOR_FRIENDLY_NAME;
+    }
+
+    /**
+     * Get the name of the Authenticator
+     */
+    @Override
+    public String getName() {
+        return LinkedInAuthenticatorConstants.LINKEDIN_CONNECTOR_NAME;
+    }
+
+    /**
+     * Get Configuration Properties
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+        List<Property> configProperties = new ArrayList<>();
+
+        Property clientId = new Property();
+        clientId.setName(LinkedInAuthenticatorConstants.CLIENT_ID);
+        clientId.setDisplayName("Client Id");
```

Comment:
Moved to constant file https://github.com/wso2-extensions/identity-outbound-auth-linkedIn/pull/3